crates/ra_mbe/src/syntax_bridge.rs

```diff
@@ -0,0 +1,113 @@
+use ra_syntax::{ast, AstNode, SyntaxNode, SyntaxKind::*};
+
+pub fn macro_call_to_tt(call: &ast::MacroCall) -> Option<tt::Subtree> {
+    let tt = call.token_tree()?;
+    convert_tt(tt.syntax())
+}
+
+fn convert_tt(tt: &SyntaxNode) -> Option<tt::Subtree> {
+    let first_child = tt.first_child()?;
+    let last_child = tt.last_child()?;
+    let delimiter = match (first_child.kind(), last_child.kind()) {
+        (L_PAREN, R_PAREN) => tt::Delimiter::Parenthesis,
+        (L_CURLY, R_CURLY) => tt::Delimiter::Brace,
+        (L_BRACK, R_BRACK) => tt::Delimiter::Bracket,
+        _ => return None,
+    };
+    let mut token_trees = Vec::new();
+    for child in tt.children().skip(1) {
+        if child == first_child || child == last_child || child.kind().is_trivia() {
```

Comment:
Doesn't this skip the inner `()` in the following code too?

```
( abc () )
```

Reply:
I think it should be handled by `let child: tt::TokenTree = if child.kind() == TOKEN_TREE`. But yeah, in general this all glosses over a **ton** of stuff. In this PR, I want to focus on creating a syntax -> tt -> mbe -> tt -> syntax interface. The `tt -> mbe` and `mbe -> tt` arrows could become `good-first-issue`s :) 

Here's a good test suite: https://github.com/intellij-rust/intellij-rust/blob/master/src/test/kotlin/org/rust/lang/core/macros/RsMacroExpansionTest.kt#L10-L680